src/gossip/service.zig

```diff
@@ -243,30 +243,26 @@ pub const GossipService = struct {
 
         // setup the threadpool for processing messages
         const n_threads: usize = @min(std.Thread.getCpuCount() catch 1, THREAD_POOL_SIZE);
-        const thread_pool = ThreadPool.init(.{
-            .max_threads = @intCast(n_threads),
-            .stack_size = 2 * 1024 * 1024,
-        });
         logger.info().logf("starting threadpool with {} threads", .{n_threads});
 
         // setup the table
         var gossip_table = try GossipTable.init(allocator, gossip_data_allocator);
         errdefer gossip_table.deinit();
 
-        // setup the active set for push messages
-        var active_set = ActiveSet.init(allocator);
-        errdefer active_set.deinit();
-
         // setup entrypoints
-        var entrypoints = ArrayList(Entrypoint).init(allocator);
-        errdefer entrypoints.deinit();
+        const entrypoints: []Entrypoint = entrypoints: {
+            const entrypoint_addrs = maybe_entrypoints orelse break :entrypoints &.{};
```

Comment:
could this block be an `allocator.dupe`?

Reply:
I had the same initial gut reaction, but it's a transformation from `[]SocketAddr` to `[]Entrypoint`.